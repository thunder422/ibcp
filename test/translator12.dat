######################################################
#  Translator Test #12: INPUT Commands (2011-03-20)  #
######################################################

INPUT A
INPUT A,B
INPUT A,B%,C$
INPUT A$;
INPUT A(I%),B%(I%);
INPUT PROMPT "Enter Number: "; N%
INPUT PROMPT A$+":"; A
INPUT PROMPT "Two Values", A, B;
INPUT PROMPT P$(I%), S$(I%)
INPUT PROMPT F$(I%), S$(I%)
INPUT PROMPT FNA$(I%), S$(I%)
# error tests
INPUT
INPUT,
INPUT;
INPUT; -A
INPUT (A)
INPUT INT(A)
INPUT LEFT$(A$,1)
INPUT FNA
INPUT FNA(X)
INPUT A B
INPUT A+B
INPUT A,
INPUT PROMPT
INPUT PROMPT A+B*C
INPUT PROMPT A$
INPUT PROMPT A$+
INPUT PROMPT A$;
# more error tests
INPUT PROMPT 1;A
INPUT PROMPT (A+B);C
INPUT PROMPT INT(A);B
