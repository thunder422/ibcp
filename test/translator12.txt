
ibcp  Copyright (C) 2010-2012  Thunder422
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to
redistribute it under certain conditions.

Table initialization successful.

Input: INPUT A
Output: InputBegin InputParse'End' A<ref> InputAssign INPUT 

Input: INPUT A,B
Output: InputBegin InputParse InputParse'End' A<ref> InputAssign B<ref> InputAssign INPUT 

Input: INPUT A,B%,C$
Output: InputBegin InputParse InputParseInt InputParseStr'End' A<ref> InputAssign B%<ref> InputAssignInt C$<ref> InputAssignStr INPUT 

Input: INPUT A$;
Output: InputBegin InputParseStr'End' A$<ref> InputAssignStr INPUT'Keep' 

Input: INPUT A(I%),B%(I%);
Output: InputBegin InputParse InputParseInt'End' I%<ref> A(<ref>[I%<ref>] InputAssign I%<ref> B%(<ref>[I%<ref>] InputAssignInt INPUT'Keep' 

Input: INPUT PROMPT "Enter Number: "; N%
Output: "Enter Number: " InputBeginStr["Enter Number: "] InputParseInt'End' N%<ref> InputAssignInt INPUT-PROMPT 

Input: INPUT PROMPT A$+":"; A
Output: A$ ":" +$[A$,":"] InputBeginTmp InputParse'End' A<ref> InputAssign INPUT-PROMPT 

Input: INPUT PROMPT "Two Values", A, B;
Output: "Two Values" InputBeginStr'Question'["Two Values"] InputParse InputParse'End' A<ref> InputAssign B<ref> InputAssign INPUT-PROMPT'Keep' 

Input: INPUT PROMPT P$(I%), S$(I%)
Output: I%<ref> P$([I%<ref>] InputBeginStr'Question'[P$(] InputParseStr'End' I%<ref> S$(<ref>[I%<ref>] InputAssignStr INPUT-PROMPT 

Input: INPUT
            ^-- expected variable


Input: INPUT,
            ^-- expected variable


Input: INPUT;
            ^-- expected variable


Input: INPUT; -A
            ^-- expected variable


Input: INPUT (A)
             ^-- expected variable


Input: INPUT INT(A)
             ^^^^-- expected variable


Input: INPUT LEFT$(A$,1)
             ^^^^^^-- expected variable


Input: INPUT FNA
             ^^^-- expected variable


Input: INPUT FNA(X)
             ^^^-- expected variable


Input: INPUT A B
               ^-- expected semicolon, comma or end-of-statement


Input: INPUT A+B
              ^-- expected semicolon, comma or end-of-statement


Input: INPUT A,
               ^-- expected variable


Input: INPUT PROMPT
                   ^-- expected string expression


Input: INPUT PROMPT A+B*C
                    ^^^^^-- expected string expression


Input: INPUT PROMPT A$
                      ^-- expected operator, semicolon or comma


Input: INPUT PROMPT A$+
                       ^-- expected string expression


Input: INPUT PROMPT A$;
                       ^-- expected variable


