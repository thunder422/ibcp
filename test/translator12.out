
Input: INPUT A
Output: INPUT A 

Input: INPUT A,B
Output: INPUT A, B 

Input: INPUT A,B%,C$
Output: INPUT A, B%, C$ 

Input: INPUT A$;
Output: INPUT A$; 

Input: INPUT A(I%),B%(I%);
Output: INPUT A(I%), B%(I%); 

Input: INPUT PROMPT "Enter Number: "; N%
Output: INPUT PROMPT "Enter Number: "; N% 

Input: INPUT PROMPT A$+":"; A
Output: INPUT PROMPT A$+":"; A 

Input: INPUT PROMPT "Two Values", A, B;
Output: INPUT PROMPT "Two Values", A, B; 

Input: INPUT PROMPT P$(I%), S$(I%)
Output: INPUT PROMPT P$(I%), S$(I%) 

Input: INPUT PROMPT F$(I%), S$(I%)
Output: INPUT PROMPT F$(I%), S$(I%) 

Input: INPUT PROMPT FNA$(I%), S$(I%)
Output: INPUT PROMPT FNA$(I%), S$(I%) 

Input: INPUT
            ^-- expected variable

Input: INPUT,
            ^-- expected variable

Input: INPUT;
            ^-- expected variable

Input: INPUT; -A
            ^-- expected variable

Input: INPUT (A)
             ^-- expected variable

Input: INPUT INT(A)
             ^^^^-- expected variable

Input: INPUT LEFT$(A$,1)
             ^^^^^^-- expected variable

Input: INPUT FNA
             ^^^-- expected variable

Input: INPUT FNA(X)
             ^^^-- expected variable

Input: INPUT A B
               ^-- expected comma, semicolon or end-of-statement

Input: INPUT A+B
              ^-- expected comma, semicolon or end-of-statement

Input: INPUT A,
               ^-- expected variable

Input: INPUT PROMPT
                   ^-- expected string expression

Input: INPUT PROMPT A+B*C
                    ^^^^^-- expected string expression

Input: INPUT PROMPT A$
                      ^-- expected operator, semicolon or comma

Input: INPUT PROMPT A$+
                       ^-- expected string expression

Input: INPUT PROMPT A$;
                       ^-- expected variable

Input: INPUT PROMPT 1;A
                    ^-- expected string expression

Input: INPUT PROMPT (A+B);C
                    ^^^^^-- expected string expression

Input: INPUT PROMPT INT(A);B
                    ^^^^^^-- expected string expression

